cmd/container-disk-v2alpha/main_test.go

```diff
@@ -0,0 +1,42 @@
+package container_disk_v2alpha_test
+
+import (
+	"flag"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var containerDiskBinary string
+
+func init() {
+	flag.StringVar(&containerDiskBinary, "container-disk-binary", "_out/cmd/container-disk-v2alpha/container-disk", "path to container disk binary")
+	flag.Parse()
+	containerDiskBinary = filepath.Join("../../", containerDiskBinary)
+}
+
+var _ = Describe("the containerDisk binary", func() {
+
+	It("should be able to handle 200 connections in 5 seconds without rejecting one of them", func() {
+		dir, err := ioutil.TempDir("", "container-disk")
+		Expect(err).ToNot(HaveOccurred())
+		defer os.RemoveAll(dir)
+		cmd := exec.Command(containerDiskBinary, "-c", filepath.Join(dir, "testsocket"))
+		Expect(cmd.Start()).To(Succeed())
+
+		time.Sleep(1 * time.Second)
+		for i := 0; i < 200; i++ {
+			conn, err := net.Dial("unix", filepath.Join(dir, "testsocket.sock"))
+			Expect(err).ToNot(HaveOccurred())
+			conn.Close()
+			time.Sleep(25 * time.Millisecond)
```

Comment:
actually, we don't want the sleep do we? we actually want to know if it can handle then all at once.

In fact, I'd make this test do all 200 Dials in a separate goroutine and use a waitgroup to wait on them exit.

Reply:
As it is right now, we sleep for 1 second (because we also want to check if the socket still exists). The rate-limit I added here ensures that we can cope with an initial delay of a second, until it finally starts accepting all the connections in a row.

Reply:
It still ensures that we definitely process socket connections way way faster than they are ever allowed to come in from virt-handler.

Reply:
as long as dial has a timeout > a few seconds, we should be able to flood connections in. right?